components/Buttons/src/MDCButton.m

```diff
@@ -923,6 +926,29 @@ - (void)updateTitleFont {
   [self setNeedsLayout];
 }
 
+- (void)setShapeGenerator:(id<MDCShapeGenerating>)shapeGenerator {
+  if (shapeGenerator) {
+    self.layer.shadowPath = nil;
+  } else {
+    self.layer.shadowPath = [self boundingPath].CGPath;
+  }
+  self.layer.shapeGenerator = shapeGenerator;
+  self.layer.colorLayer.zPosition = -1;
```

Comment:
Instead of setting the zPosition I had moved the colorLayer to be at its rightful place by removing and adding it below the inkView and the button's imageView. Also have written an explanation as a comment elaborating why